src/crimson/osd/scheduler/mclock_scheduler.h

```diff
@@ -15,112 +15,111 @@
 
 #pragma once
 
+#include <functional>
 #include <ostream>
 #include <map>
 #include <vector>
 
 #include "boost/variant.hpp"
 
 #include "dmclock/src/dmclock_server.h"
-
 #include "crimson/osd/scheduler/scheduler.h"
+#include "crimson/mon/MonClient.h"
+
 #include "common/config.h"
 #include "common/ceph_context.h"
-
+#include "common/mclock_common.h"
 
 namespace crimson::osd::scheduler {
 
-using client_id_t = uint64_t;
-using profile_id_t = uint64_t;
-
-struct client_profile_id_t {
-  client_id_t client_id;
-  profile_id_t profile_id;
-  auto operator<=>(const client_profile_id_t&) const = default;
-};
-
-
-struct scheduler_id_t {
-  scheduler_class_t class_id;
-  client_profile_id_t client_profile_id;
-  auto operator<=>(const scheduler_id_t&) const = default;
-};
-
 /**
  * Scheduler implementation based on mclock.
  *
  * TODO: explain configs
  */
 class mClockScheduler : public Scheduler, md_config_obs_t {
 
-  class ClientRegistry {
-    std::array<
-      crimson::dmclock::ClientInfo,
-      static_cast<size_t>(scheduler_class_t::client)
-    > internal_client_infos = {
-      // Placeholder, gets replaced with configured values
-      crimson::dmclock::ClientInfo(1, 1, 1),
-      crimson::dmclock::ClientInfo(1, 1, 1)
-    };
+  crimson::common::CephContext *cct;
+  unsigned cutoff_priority;
+  PerfCounters *logger;
 
-    crimson::dmclock::ClientInfo default_external_client_info = {1, 1, 1};
-    std::map<client_profile_id_t,
-	     crimson::dmclock::ClientInfo> external_client_infos;
-    const crimson::dmclock::ClientInfo *get_external_client(
-      const client_profile_id_t &client) const;
-  public:
-    void update_from_config(const ConfigProxy &conf);
-    const crimson::dmclock::ClientInfo *get_info(
-      const scheduler_id_t &id) const;
-  } client_registry;
+  ClientRegistry client_registry;
+  MclockConfig mclock_conf;
 
   using mclock_queue_t = crimson::dmclock::PullPriorityQueue<
     scheduler_id_t,
     item_t,
     true,
     true,
     2>;
+  using priority_t = unsigned;
+  using SubQueue = std::map<priority_t,
+	std::list<item_t>,
+	std::greater<priority_t>>;
   mclock_queue_t scheduler;
-  std::list<item_t> immediate;
+  /**
+   * high_priority
+   *
+   * Holds entries to be dequeued in strict order ahead of mClock
+   * Invariant: entries are never empty
+   */
+  SubQueue high_priority;
+  priority_t immediate_class_priority = std::numeric_limits<priority_t>::max();
 
   static scheduler_id_t get_scheduler_id(const item_t &item) {
     return scheduler_id_t{
       item.params.klass,
-	client_profile_id_t{
-	item.params.owner,
-	  0
-	  }
+      client_profile_id_t{
```

Comment:
Use the default `client_profile_id_t()` here as in classic. This allows us to group items having different `owner` ids but belonging to a specific SchedulerClass as a single client.

The usage of different 'owner` ids comes into the picture when we want to implement QoS for individual external clients.